modules/unsupported/wfs-ng/src/main/java/org/geotools/data/wfs/internal/parsers/AbstractGetFeatureResponseParserFactory.java

```diff
@@ -92,134 +77,25 @@ public boolean canProcess(final WFSRequest request, final String contentType) {
         return matches;
     }
 
-    /**
-     * Returns either a {@link FeatureCollectionParser} or an {@link ExceptionReportParser}
-     * depending on what the server returned.
-     * <p>
-     * Ideally, the decision should only be taken based on the WFS response's content-type HTTP
-     * header. Truth is, some WFS implementations does not set proper HTTP response headers so a bit
-     * of an heuristic may be needed in order to identify the actual response.
-     * </p>
-     * 
-     * @see WFSResponseFactory#createParser(WFSResponse)
-     * @see FeatureCollectionParser
-     * @see ExceptionReportParser
-     */
-    public WFSResponse createResponse(WFSRequest request, HTTPResponse response) throws IOException {
-
-        // We can't rely on the server returning the correct output format. Some, for example
-        // CubeWerx, upon a successful GetFeature request, set the response's content-type
-        // header to plain "text/xml" instead of "text/xml;subtype=gml/3.1.1". So we'll do a bit
-        // of heuristics to find out what it actually returned
-        final int buffSize;
-        if (LOGGER.isLoggable(Level.FINER)) {
-            buffSize = 4096;
-        } else {
-            buffSize = 512;
-        }
-        PushbackInputStream pushbackIn = new PushbackInputStream(response.getResponseStream(), buffSize);
```

Comment:
Looks like this logic moved to AbstractWFSResponseFactory.java below

Reply:
Yes, right.